### DISCUSSION TITLE
Can I disable highlighting for parts of a string?

### DISCUSSION THREAD

**Discussion by Hubro** [Textualize/rich#2794] [Q&A]
I'm using Rich to format my log messages. This works nicely, and I love the automatic highlighting, but I wish I could disable highlighting for the parts of the line that I have already manually styled with [console markup](https://rich.readthedocs.io/en/latest/markup.html).

Could there be a console markup tag to conditionally enable or disable highlighting for a chunk of text?

For example:

    [bold blue]{loglevel}[/] [i]{time}[/] [highlight]{message}[/]

Basically I only want Rich to highlight the *message* part of my log lines.

Is there currently any way to achieve this?

**Comment by willmcgugan** [comment]
Yes, you can construct text objects with `Text.from_markup` and you can highlight text with a highlighter. You can then combine the text in any arrangement you like. See the following pages from the docs:

https://rich.readthedocs.io/en/latest/text.html
https://rich.readthedocs.io/en/latest/highlighting.html
